network/p2p/p2p_sync.go

```diff
@@ -188,8 +188,8 @@ func allPeersFilter(id peer.ID) bool {
 
 func waitSubsetOfPeers(
 	logger *zap.Logger,
-	getSubsetOfPeers func(logger *zap.Logger, vpk spectypes.ValidatorPK, maxPeers int, filter func(peer.ID) bool) (peers []peer.ID, err error),
-	vpk spectypes.ValidatorPK,
+	getSubsetOfPeers func(logger *zap.Logger, senderID []byte, maxPeers int, filter func(peer.ID) bool) (peers []peer.ID, err error),
```

Comment:
we should rename all occurrences of senderID to dutyExecutorID and so on (because the spec recently made this change)

the reason i'm not doing this now is to lessen the number of conflicts with other PRs pending merge to alan/no-fork

it's likely better to do this refactor after merging message validation and at least part of fork support